src/vivarium/framework/results/context.py

```diff
@@ -253,17 +262,15 @@ def register_observation(
                 f"Observation name '{name}' is already used: {self.observations[name]}."
             )
 
-        # Instantiate the observation and add it and its (pop_filter, stratifications)
-        # tuple as a key-value pair to the self.observations[when] dictionary.
         observation = observation_type(
             name=name,
-            pop_filter=pop_filter,
+            population_filter_details=population_filter_details,
             when=when,
             requires_attributes=requires_attributes,
             **kwargs,
         )
         self.observations[name] = observation
-        self.grouped_observations[observation.when][observation.pop_filter][
+        self.grouped_observations[observation.when][population_filter_details][
```

Comment:
Nit: it's possible (though unlikely) that the constructor of the observation might mutate the population filter. Isn't it safer to use `observation.population_filter_details` here?